File: z-clients/javascript/src/apis/rateLimitNamespace.ts
// this file is @generated

import {
    type RateLimitConfigureNamespaceIn,
    RateLimitConfigureNamespaceInSerializer,
} from '../models/rateLimitConfigureNamespaceIn';
import {
    type RateLimitConfigureNamespaceOut,
    RateLimitConfigureNamespaceOutSerializer,
} from '../models/rateLimitConfigureNamespaceOut';
import {
    type RateLimitGetNamespaceIn,
    RateLimitGetNamespaceInSerializer,
} from '../models/rateLimitGetNamespaceIn';
import {
    type RateLimitGetNamespaceOut,
    RateLimitGetNamespaceOutSerializer,
} from '../models/rateLimitGetNamespaceOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class RateLimitNamespace {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Configure rate limiter namespace */
    public configure(
        rateLimitConfigureNamespaceIn: RateLimitConfigureNamespaceIn,
    ): Promise<RateLimitConfigureNamespaceOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.rate-limit.namespace.configure");

        request.setBody(
            RateLimitConfigureNamespaceInSerializer._toJsonObject(rateLimitConfigureNamespaceIn)
        );
        
        return request.send(
            this.requestCtx,
            RateLimitConfigureNamespaceOutSerializer._fromJsonObject,
        );
    }/** Get rate limiter namespace */
    public get(
        rateLimitGetNamespaceIn: RateLimitGetNamespaceIn,
    ): Promise<RateLimitGetNamespaceOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.rate-limit.namespace.get");

        request.setBody(
            RateLimitGetNamespaceInSerializer._toJsonObject(rateLimitGetNamespaceIn)
        );
        
        return request.send(
            this.requestCtx,
            RateLimitGetNamespaceOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/rust/src/api/rate_limit_namespace.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct RateLimitNamespace<'a> {
    cfg: &'a Configuration,
}

impl<'a> RateLimitNamespace<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Configure rate limiter namespace
    pub async fn configure(
        &self,
        rate_limit_configure_namespace_in: RateLimitConfigureNamespaceIn,
    ) -> Result<RateLimitConfigureNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.rate-limit.namespace.configure")
            .with_body(rate_limit_configure_namespace_in)
            .execute(self.cfg)
            .await
    }

    /// Get rate limiter namespace
    pub async fn get(
        &self,
        rate_limit_get_namespace_in: RateLimitGetNamespaceIn,
    ) -> Result<RateLimitGetNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.rate-limit.namespace.get")
            .with_body(rate_limit_get_namespace_in)
            .execute(self.cfg)
            .await
    }
}
